### ISSUE TITLE
False positive CS8602 when using Extension Member Feature

### ISSUE THREAD

**Issue by fitdev** [dotnet/roslyn#78182]
**Version Used**: VS 2022 17.14.P3

**Steps to Reproduce**:

```cs
static partial class Strings {
  extension([MaybeNullWhen(true), NotNullWhen(false)] string? s) {
    [DebuggerStepThrough]
    public bool IsEmpty() => string.IsNullOrEmpty(s);
}
class c {
  public string? TextKey {
    get;
    set;
  }
  bool IsValid => !TextKey.IsEmpty(); // <- CS8602: Dereference of a possibly null reference.
}
```
This is a regression introduced with new extension members feature. Since the old version like:

```cs
static partial class Strings {
    [DebuggerStepThrough]
    public static bool IsEmpty([MaybeNullWhen(true), NotNullWhen(false)] this string? s) => string.IsNullOrEmpty(s);
}
```
did not produce this false positive.

**Diagnostic Id**:

CS8602: Dereference of a possibly null reference.

**Expected Behavior**:
Dereference is safe, no warning.

**Actual Behavior**:
False positive warning.

**Comment by dougcunha** [comment]
Was this supposed to be fixed? I've just started to convert a bunch of extension methods to extensions members, and I'm getting a lot of false positive warnings related to nullable when converting to property extension member.

Visual Studio Version 17.14.12

```
public static class DateTimeExtensions
{
    extension(DateTime? @this)
    {
        /// <summary>
        /// Returns null if the DateTime is equal to DateTime.MinValue (0001-01-01T00:00:00).
        /// </summary>
        public DateTime? NullIfZero
            => @this == DateTime.MinValue ? null : @this;
    }
}
```

<img width="640" height="229" alt="Image" src="https://github.com/user-attachments/assets/f4c66488-dfe0-4289-891b-09b152a8d97e" />
